### ISSUE TITLE
No exception handling

### ISSUE THREAD

**Issue by karlvlam** [seratch/AWScala#139]
Currently the library just return None for catched exception, which is super difficult to debug. It is suggested to change to return the error to the caller.

**Comment by seratch** [comment]
which methods?

**Comment by adnanced** [comment]
Yes, I have the same problems. Something wrong is going on in only one of my environments and I have no idea what. I checked some relevant functions and saw that all of those that wrap around java implementation are ignoring or throwing exceptions (at best). That shouldn't happen in Scala as that is one of the benefits of this language and functional programming. 
If you need examples check functions: `CredentialsLoader.load()`, `S3.buckets()`, `S3.location()`, `S3.deleteBucket()` and etc. It's really a lot of them...
Most of them should have return type of `Try[T]` or `Future[T]`. Fixing this issue would change a lot in the library and it's API (breaking changes), so I can understand if you don't want to go by this path.
